src/api/localStorage.js

```diff
@@ -0,0 +1,97 @@
+export class LocalStorage {
+  constructor() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  createUser(user) {
+    this.updateUsers();
+    user.id = this.generateId();
+    this.users.push(user);
+    localStorage.setItem('UserAccounts', JSON.stringify(this.users));
+    console.log('User Added: ', user);
+  } 
+
+  updateUsers() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+ 
+  getUser(id) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    console.log('getUser() -> User Found: ', user);
+    return user;
+  }
+
+  getAllUsers() {
+    console.log('getAllUsers() -> Users Found: ', this.users);
+    return this.users;
+  }
+
+  editUser(id, editName, editValue) {
+    const editObj = {};
+    editObj[`${editName}`] = `${editValue}`;
+    const updatedUser = Object.assign(this.getUser(id), editObj);
+    console.log('Updated User: ', updatedUser);
+    this.createUser(updatedUser);
+  }
+
+  replaceUser(id, update) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    const replacedUser = Object.assign(user, update);
+    console.log('replaceUser() -> Replaced User: ', replacedUser)
+    return replacedUser;
+  }
+
+  deleteUser(id) {
+    this.updateUsers();
+    this.users.forEach((user, index) => {
+      if (String(user.id) === String(id)) {
+        this.users.splice(index, 1);
+        localStorage.setItem('UserAccounts', JSON.stringify(this.users));
```

Comment:
You could use your new helper function here

Reply:
Completed